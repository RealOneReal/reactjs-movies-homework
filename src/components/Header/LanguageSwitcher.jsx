import { FormControl, Select, MenuItem } from "@mui/material";
import { useState } from "react";
import { useDispatch } from "react-redux";
import { language } from "../../redux/search/searchSlice";

const LanguageSwitcher = () => {
  const [languageFromInput, setLanguage] = useState("EN");
  const dispatch = useDispatch();

  const handleChange = ({ target }) => {
    setLanguage(target.value);
    dispatch(language(target.value));
  };

  const styles = {
    menuItem: {
      color: "secondary.dark",
    },
    select: {
      color: "primary.dark",
      "& .MuiSvgIcon-root": {
        color: "primary.dark",
      },
    },
  };

  return (
    <FormControl>
      <Select
        value={languageFromInput}
        onChange={handleChange}
        variant="standard"
        color="secondary"
        sx={styles.select}
      >
        <MenuItem value="EN" sx={styles.menuItem}>
          EN
        </MenuItem>
        <MenuItem value="RU" sx={styles.menuItem}>
          RU
        </MenuItem>
      </Select>
    </FormControl>
  );
};

export default LanguageSwitcher;
