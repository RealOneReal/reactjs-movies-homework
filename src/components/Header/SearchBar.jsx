import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Box, TextField, IconButton } from "@mui/material";
import SearchIcon from "@mui/icons-material/Search";
import { inputSearch } from "../../redux/search/searchSlice";
import {
  fetchMoviesBySearch,
  fetchGenresConfig,
} from "../../redux/movies/moviesSlice";

const SearchBar = () => {
  const [value, setValue] = useState("");
  const dispatch = useDispatch();
  const { language } = useSelector((state) => state.search);

  const handleInput = ({ target }) => {
    setValue(target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    event.key === "Enter " && dispatch(inputSearch(value));
    dispatch(inputSearch(value));
    value && dispatch(fetchGenresConfig({ language }));
    value && dispatch(fetchMoviesBySearch({ value, pageNumber: 1, language }));
  };

  const styles = {
    formGroup: {
      width: { md: "400px", xs: "200px" },
      display: "flex",
      flexDirection: "row",
      columnGap: "8px",
    },
    textField: {
      width: {
        xs: "150px",
        md: "350px",
      },
      "& .MuiInputBase-input": {
        color: "primary.dark",
      },
    },
    iconButton: { width: "20px", color: "primary.dark" },
  };

  return (
    <Box sx={styles.formGroup} component="form" onSubmit={handleSubmit}>
      <TextField
        value={value}
        onChange={handleInput}
        type="text"
        placeholder="Movies,person,movie theaters"
        variant="standard"
        sx={styles.textField}
      />
      <IconButton sx={styles.iconButton} type="submit">
        <SearchIcon />
      </IconButton>
    </Box>
  );
};

export default SearchBar;
