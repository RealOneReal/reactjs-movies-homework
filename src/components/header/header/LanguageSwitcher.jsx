import { FormControl, Select, MenuItem } from "@mui/material";
import { useState } from "react";

const LanguageSwitcher = () => {
  const [language, setLanguage] = useState("EN");

  const handleChange = (event) => {
    setLanguage(event.target.value);
  };

  return (
    <FormControl>
      <Select value={language} onChange={handleChange} variant="standard">
        <MenuItem value="EN">EN</MenuItem>
        <MenuItem value="RU">RU</MenuItem>
      </Select>
    </FormControl>
  );
};

export default LanguageSwitcher;
