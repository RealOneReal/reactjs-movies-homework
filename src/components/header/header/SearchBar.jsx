import { TextField } from "@mui/material";
import { useState } from "react";

const SearchBar = () => {
  const [value, setValue] = useState("");

  const handleInput = (event) => {
    setValue(event.target.value);
  };

  const styles = {
    textField: {
      width: "400px",
    },
  };

  return (
    <TextField
      value={value}
      onChange={handleInput}
      type="search"
      placeholder="Movies,person,movie theaters"
      variant="standard"
      sx={styles.textField}
    />
  );
};

export default SearchBar;
