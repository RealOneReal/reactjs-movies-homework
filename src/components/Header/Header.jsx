import { AppBar, Toolbar, Typography } from "@mui/material";
import SearchBar from "./SearchBar";
import LanguageSwitcher from "./LanguageSwitcher";

const styles = {
  toolbar: { display: "flex", justifyContent: "space-evenly" },
  typography: { fontWeight: 600, color: "primary.dark" },
};

const title = "TITLE";

const Header = () => {
  return (
    <AppBar position="static" sx={{ backgroundColor: "secondary.main" }}>
      <Toolbar sx={styles.toolbar}>
        <Typography variant="h4" component="h1" sx={styles.typography}>
          {title}
        </Typography>
        <SearchBar />
        <LanguageSwitcher />
      </Toolbar>
    </AppBar>
  );
};

export default Header;
