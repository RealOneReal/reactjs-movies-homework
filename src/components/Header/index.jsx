import { AppBar, Toolbar, Typography } from "@mui/material";
import SearchBar from "./SearchBar";
import LanguageSwitcher from "./LanguageSwitcher";

const styles = {
  toolbar: { display: "flex", justifyContent: "space-evenly" },
  typography: { fontWeight: 600 },
};

const Header = () => {
  return (
    <AppBar position="static">
      <Toolbar sx={styles.toolbar}>
        <Typography variant="h4" component="h1" sx={styles.typography}>
          TITLE
        </Typography>
        <SearchBar />
        <LanguageSwitcher />
      </Toolbar>
    </AppBar>
  );
};

export default Header;
