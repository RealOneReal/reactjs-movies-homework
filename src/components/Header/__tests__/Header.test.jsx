import renderer from "react-test-renderer";
import SearchBar from "../SearchBar";
import LanguageSwitcher from "../LanguageSwitcher";
import Header from "../Header";

jest.mock("../SearchBar", () => {
  return "SearchBar";
});

jest.mock("../LanguageSwitcher", () => {
  return "LanguageSwitcher";
});

describe("Header", () => {
  it("should render", () => {
    const header = renderer.create(<Header />);
    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
