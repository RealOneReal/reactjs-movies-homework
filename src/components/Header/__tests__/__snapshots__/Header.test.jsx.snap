// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-z6o611-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-80fgyj-MuiToolbar-root"
  >
    <h1
      className="MuiTypography-root MuiTypography-h4 css-1uoyj95-MuiTypography-root"
    >
      TITLE
    </h1>
    <SearchBar />
    <LanguageSwitcher />
  </div>
</header>
`;
